boards/arm/mps2_an521/mps2_an521_nonsecure.dts

```diff
@@ -45,6 +45,11 @@
 		reg = <0x28100000 0x100000>;
 	};
 
+	sramx: memory@28180000 {
+		compatible = "mmio-sram";
+		reg = <0x28180000 0x8000>;
+	};
+
```

Comment:
Very interesting implementation based on the memory declaration in the DTS. I hesitated to do same in my pull request (https://github.com/zephyrproject-rtos/zephyr/pull/16985).
A drawback in this implementation is that the memory region is project dependent. It is needed only for OpenAMP, but declared for a board, so for all projects.
For instance, this result here in an overlap between the memory declared for the zephyr firmware and this shared memory. not safe...
An other alternative would be  to limit the zephyr SRAM, but impact is for all projects...

So a point to solve here (IMHO) is to make this shared region optional ( project dependent) to allow to use it for some other purposes. But also to configure the rest of the ram according to this shared memory.

To get around the issue, In my pull request, i have implemented the memory pool  as a linker memory region, under CONFIG_OPENAMP. But I'm not satisfied with my implementation. If my shared memory is optional, This does not prevent overlap with the zephyr SRAM. 

I have no real solution  to propose:), i can just try to clarify the equation:
- There is a dependency on project: use of a CONFIG_OPENAMP?
- There is a dependency on the platform ( memory region definition + Zephyr RAM resizing): DTS overlay, linker with some CONFIG values, ...?